### ISSUE TITLE
False error on kebab-case files

### ISSUE THREAD

**Issue by cajoy** [sindresorhus/eslint-plugin-unicorn#343]
I have file with name like [id].tsx

and getting following error:

```
/pages/[id].tsx
  1:1  error  Filename is not in kebab case. Rename it to `id.tsx`  unicorn/filename-case
```

**Comment by MrHen** [comment]
Just to confirm, your file is literally the characters `[id].tsx`? With the square brackets included?

**Comment by futpib** [comment]
This is probably related to how next.js does routing https://github.com/zeit/next.js/tree/ca13752e240a85f8af7a875a6dba0a515c6d8011/examples/dynamic-routing/pages/post/[id]

**Comment by MrHen** [comment]
Ah, interesting.

As for the rule, strictly speaking, `[id].tsx` _isn't_ in kebab-case. I haven't looked at the code yet but I'm guessing the regex involved is only allowing lowercase letters, numbers and hyphens. We could try to:

1. Add more allowed symbols
2. Add a configuration that allows the format used by next.js
3. Add a custom regex option that lets you add whatever exceptions you want
4. Ignore this use case and either disable the rule or add an exclusion at the top of the file

**Comment by jaythomas** [comment]
What if `filename-case` just exposed a regex property you can set to whitelist patterns?

**Comment by sindresorhus** [comment]
Let's do `1`. I can't imagine anyone would accidentally wrap a filename in `[]`. It's intentional then. The point of the rule is to catch actual naming mistakes. So we'll just strip leading `[` and trailing `]` before checking the filename.

**Comment by sindresorhus** [comment]
@jaythomas We can do that in addition to `1` if you can provide other use-cases where you would need it.

**Comment by jaythomas** [comment]
@sindresorhus allow me to provide 3 of my use cases.

1. Although all my files are kebab-case, some files have a special suffix such as `interface_spec.js` or `interface_snap.js` and the rule thinks these are snakecased.

2. A few times I like to break the kebab-case rule to that the filename matches the factory name, such as `$logger.js` so I'd like to account for these (until I can fix the factory names at least).

3. On another large project is I have legacy code folders where the files have weird names that don't match the kebab-case the rest of the project uses so if I were able to pass the rule a regex I could tell it to either match kebab-case OR if the file/directory name contains "^(apr|cgi|conf)" or any of my app's internal identifiers used on older files.

The first two issues could yet more edge case handled by the filename-case rule but I don't think they necessarily should because on personal projects I prefer the strictness and predictability of the rule.
The third use case I could maybe create eslintrc files in each subfolder with weird filenames, but that disables the rule for all files in the folder.

The regex solution could handle these 3 situations elegantly from my point of view. If such an enhancement was welcome I may be able to contribute a pull request.

**Comment by fisker** [comment]
How about split filename to words and check one by one? 

Like I did in `prevent-abbreviations` for filenames

https://github.com/sindresorhus/eslint-plugin-unicorn/blob/91bbff52d534908a3688817f655af91f7920c985/rules/prevent-abbreviations.js#L272

maybe something like 

```js
`[name-with-dash~~~~~~~$dollor-prefixed~~~~~name_with_underscore].tsx`
  .split(/[^a-z-_$]/i)
  .filter(Boolean)
```
